### ISSUE TITLE
GUAC GraphQL vs. PostgreSQL TLS

### ISSUE THREAD

**Issue by otaviof** [trustification/trustification#1132]
Setting `openshift.useServiceCa` as `true` adds the following environment variables on the POD ([1](https://github.com/trustification/trustification/blob/7b62f2650b6b85783fc7adb75f1e426701007b27/deploy/k8s/charts/trustification/templates/services/guac/graphql/030-Deployment.yaml#L32-L43
), [2](https://github.com/trustification/trustification/blob/7b62f2650b6b85783fc7adb75f1e426701007b27/deploy/k8s/charts/trustification/templates/services/guac/graphql/030-Deployment.yaml#L52-L65)):

https://github.com/trustification/trustification/blob/f07ccfe8eaf287bc7971bbf7943e83d4ea660e7b/deploy/k8s/charts/trustification/templates/helpers/_guac.tpl#L21-L27

Having `GUAC_GQL_TLS_{CERT,KEY}_FILE` set forces the database connection to use TLS, thus `TC_PGSSLMODE` is ignored.  The value for `TC_PGSSLMODE` comes from `guac.database.sslMode` but conflicts with `openshift.useServiceCa`.

Additionally, it's important to make a distinction between these two settings, while `useServiceCa` coordinates the incoming traffic (using the same TLS configuration than a service), `TC_PGSSLMODE` controls the outgoing database connection, these two should not create side effects.

This issue is related to PR #1129 and [RHTAPBUGS-1146](https://issues.redhat.com/browse/rhtapbugs-1146).

/cc @Roming22, @flacatus, @xinredhat

**Comment by otaviof** [comment]
@dejanb, @mrizzi, that's the issue we've just discussed, please consider. Thanks in advance!

**Comment by ctron** [comment]
@dejanb @mrizzi please correct me if I am wrong, but there's not connect between `TC_PGSSLMODE` and the endpoint certs for `GUAC_GQL_TLS_`.

**Comment by dejanb** [comment]
@ctron As far as I can see so far there shouldn't be ... I need to test it to see if there's some maybe ent side effect we are not aware of

**Comment by dejanb** [comment]
@otaviof I can't reproduce this issue. Our Openshift (without AWS) deployment is exactly that scenario

https://github.com/trustification/trustification/blob/main/deploy/k8s/DEPLOYING.md#openshift-without-aws

When I try it I get TLS variables set, process started with `sslmode=disable` and everything seems to be working fine.

```

{"level":"info","ts":1709296872.4090152,"caller":"backend/migrations.go:74","msg":"skipping ent migrations"}
{"level":"info","ts":1709296872.4159927,"caller":"cmd/server.go:109","msg":"connect to https://localhost:8080/ for GraphQL playground"}
{"level":"info","ts":1709296872.416022,"caller":"cmd/server.go:113","msg":"starting server"}
{"level":"info","ts":1709296872.4160967,"caller":"cli/init.go:84","msg":"Prometheus server is listening on: 9010"}
```

There are TLS errors in the service log, like

```
2024/03/01 12:41:21 http: TLS handshake error from 10.128.4.2:39670: EOF
2024/03/01 12:41:31 http: TLS handshake error from 10.128.4.2:58506: EOF
2024/03/01 12:41:41 http: TLS handshake error from 10.128.4.2:41074: EOF
2024/03/01 12:41:51 http: TLS handshake error from 10.128.4.2:50986: EOF
```

but those are due to the health check issue. I'll tackle that next as we need to get those in order.

Also, note that we pushed one more fix in this area this morning #1130 so you might have been affected by that?

Anyhow, any more info on how to reproduce the issue and what exact error messages you are seeing (in which pods) would be helpful.

**Comment by otaviof** [comment]
@ctron, @dejanb, @alanfx thanks for the pointers 🙇 The issue I found was solved by #1131, the rollout problems were solved, and working in combination with `openshift.useServiceCa` enabled.

I'm closing this issue.
